### PR TITLE
Added 'get resource count' endpoint to resource_router

### PR DESCRIPTION
**Pull request by mtkinit**

### CODE REVIEW

File: src/routers/resource_router.py [hunk 1]
```diff
@@ -122,6 +122,13 @@ def create(self, engine: Engine, url_prefix: str) -> APIRouter:
             name=f"List {self.resource_name_plural}",
             **default_kwargs,
         )
+        router.add_api_route(
+            path=f"{url_prefix}/count/{self.resource_name_plural}/{version}",
```
Comment (josvandervelde) [inline src/routers/resource_router.py:126]: I would propose to changing the `/{version}` into a hardcoded "/0".

The `version` variable will change when the metadata will change. But the `/count` will not change then. So we should probably have a different versioning for the `/count` than for the other routes. Creating a system for that is a bit too much overengineering for now, I'd say, so let's just hardcode it for now.

Then we will have the problem of two routers defining the same route, if we would have 2 different versions running on the same time. Luckily, only one would not be deprecated. So I'd suggest adding a `if self.deprecated_from is not None` around this route.

**Review by josvandervelde — COMMENTED** [review]
Thank you very much!

Looks good. Two more minor points:
- Could you create a small testcase? You could just copy `test_router_get_all.py`, change the `GET` request, and just assert that the count is 2. That would be enough testing, I think. I could create one as well, just let me know
- Could you install pre-commit (`pre-commit install`)? Then, for every commit you make, the code style etc. will be fixed automatically

File: src/routers/resource_router.py (hunk 1 above)
Comment (mtkinit) [reply src/routers/resource_router.py:126]: Do you mean "/0" or "/v0" so that the convention stays preserved?

**Comment by mtkinit** [comment]
> Done. Thanks for recommendations, actually I just build the container, I first did not install the dependencies locally, but I did it now. On WSL2 there were some issues with mysql headers, I'll hopefully create an issue with a proposed solution in the future.

File: src/routers/resource_router.py (hunk 1 above)
Comment (josvandervelde) [reply src/routers/resource_router.py]: Good point, `/v0` is better!

Comment (mtkinit) [reply src/routers/resource_router.py]: @josvandervelde it's done.

**Review by josvandervelde — APPROVED** [review]
Excellent! Feel free to merge it

**Pull request merged**